### PR TITLE
PD 181: Improve volume inspect

### PR DESCRIPTION
**Pull request by andreipope**

### CODE REVIEW

**Comment by vilasd** [comment]
@sangleganesh can you please review this doc PR and give more details on some of the fields ?

File: content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+- __Format__: the file system used to store data. Currently, _Portworx_ supports `xfs` and `ext4`.
+- __HA__: represents the replication factor for the volume. As an example, if a volume has a replication factor of 3, it means the data is protected on 3 separate nodes.
+
+ You can set the replication factor while creating the volume, by running the  `pxctl volume create` command and passing it the `--repl` flag:
+
+ ```text
+ pxctl volume create testVol --repl=2
+ ```
+
+ For applications that require node level availability and read parallelism across nodes, _Portworx_ recommends setting a replication factor of 2 or 3. You can further increase the replication factor of a volume by running the `pxctl volume ha-update` and passing it the following flags:
```
Comment (nathan-px) [inline content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md]: We should mention the max replication factor. Saying you can *further* increase the replication factor immediately after repl3 makes it seem like you can go higher than that.

File: content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+ ```text
+ pxctl volume create --io_priority high volume-name
+ ```
+
+ See the [class-of-service](/concepts/class-of-service) page to get a better understanding of how performance levels work in _Portworx_. Additionally, note that _Portworx_ provides an easy way to [manage storage pools](/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/maintenance-mode/#storage-pool-maintenance) through the `pxctl service pool` command.
+
+- __Creation time__: indicates the creation date and time of the volume.
+- __Shared__: this field tells whether the volume is `shared` or not. A shared volume is available to multiple containers running on different hosts at the same time. See the [shared volumes](/concepts/shared/shared-volumes) page for more details
+- __Status__: indicates the status of the volume. //TODO: Needs details. What does it mean? Possible values? Just `up` and `down`?
```
Comment (nathan-px) [inline content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md]: @sangleganesh

File: content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+- __State__: shows whether the volume is `detached` or `attached`. //TODO: Needs more details. What exactly `detached` and `attached` mean? How and when is a volume expected move from one state to the other?
```
Comment (nathan-px) [inline content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md]: @sangleganesh //TODO: Needs more details. What exactly `detached` and `attached` mean? How and when is a volume expected move from one state to the other?

File: content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+- __Reads__: the number of `read` operations served by the volume.
+- __Reads MS__: the total amount of time spent doing reads, expressed in milliseconds.
+- __Bytes Read__: measures the total number of bytes read from the volume.
+- __Writes__: the number of `write` operations served by the volume.
+- __Writes MS__: the total amount of time spent doing write operations, expressed in milliseconds.
+- __Bytes Written__: represents the total amount of bytes written to the volume.
+- __IOs in progress__: tells the number of IO operations currently in progress.
+- __Bytes used__: indicates the amount of space used on the volume, expressed in KiB.
+- __Replica sets on nodes__: the list of nodes where _Portworx_ has placed volume replicas. Note that the maximum replication factor is 3 and can only be incremented or decremented **sequentially**. //TODO: This is confusing. Why can we run ` pxctl volume ha-update` with a comma-separated list of volumes then?
```
Comment (nathan-px) [inline content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md]: @sangleganesh //TODO: This is confusing. Why can we run ` pxctl volume ha-update` with a comma-separated list of volumes then?

File: content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+    Bytes Written        :  0
+    IOs in progress      :  0
+    Bytes used           :  340 KiB
+    Replica sets on nodes:
+        Set 0
+          Node          : 70.0.29.70 (Pool 0)
+    Replication Status     :  Detached
+```
+
+## Inspect volumes - JSON format
```
Comment (nathan-px) [inline content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md]: @sangleganesh I kind of think this should be deleted. Do we want to expose and talk about all of these in detail? This seems like overkill now, especially with our JQ docs covering some of the more common use-cases.

File: content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md (hunk 2 above)
Comment (sangleganesh) [reply content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md]: status is Up is volume is created successfully, or is Pending if volume/snapshot is being created.

File: content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md (hunk 3 above)
Comment (sangleganesh) [reply content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md]: State: Attached -> Volume is attached on a node in the cluster and I/O can be done on the volume.
Detached -> Volume is not being used and I/O cannot be done on the volume.
When container start using the volume, they attach and mount the volume and then do I/O. When the container is removed the corresponding volumes are unmounted and detached.

File: content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md (hunk 4 above)
Comment (sangleganesh) [reply content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md]: Can just remove the line. pxctl volume ha-update accepts only one volume, it accepts a list of nodes/pools to select where to place the new replica(s). Replication level can be changed by only 1 at a time.

File: content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md (hunk 5 above)
Comment (sangleganesh) [reply content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md]: yes we can remove it.

File: content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md (hunk 1 above)
Comment (andreipope) [reply content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md]: @nathan-px It's updated. Thanks!

File: content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md (hunk 4 above)
Comment (andreipope) [reply content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md]: great, thanks!

File: content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md (hunk 5 above)
Comment (andreipope) [reply content/reference/CLI/create-and-manage-volumes/inspect-volumes/_index.md]: I've just removed it.

File: content/reference/CLI/create-and-manage-volumes/_index.md [hunk 6]
```diff
@@ -64,48 +65,48 @@ Use "pxctl volume [command] --help" for more information about a command.
...
 
-* Durability: Set the replication level through policy, using the High Availability setting.
-* Each write is synchronously replicated to a quorum set of nodes.
-* Any hardware failure means that the replicated volume has the latest acknowledged writes.
-* Elastic: Add capacity and throughput at each layer, at any time.
-* Volumes are thinly provisioned, only using capacity as needed by the container.
-* You can expand and contract the volume’s maximum size, even after data has been written to the volume.
+* Durability: Set the replication level through policy, using the `High Availability` setting. See the [storage policy](/reference/cli/storagepolicy) page for more details.
+* Portworx synchronously replicates each write operation to a quorum set of nodes.
+* If a hardware failure occurs, the replicated volume has the latest acknowledged writes.
```
[the remark this answers is not in the document]
Comment (andreipope) [reply content/reference/CLI/create-and-manage-volumes/_index.md]: Removed.

**Comment by andreipope** [comment]
Hey, @sangleganesh I've incorporated your feedback and solved the conflicts. Please let me know if you require any other changes. Thanks!

**Pull request merged**